spec/classes/init_spec.rb

```diff
@@ -137,68 +137,117 @@
-          it { is_expected.not_to create_reboot_notify('selinux') }
-        end
-        context 'disabled -> disabled' do
-          let(:facts) do
-            os_facts = os_facts.dup
-            os_facts[:selinux] = false
-            os_facts
+          context 'ensure -> false' do
+            let(:facts) do
+              os_facts
+            end
+            let(:params) {{
+              ensure: false,
+              kernel_enforce: true
+            }}
+            it { is_expected.to compile.with_all_deps }
+            it { is_expected.to create_kernel_parameter('selinux').with_value(0).that_notifies('Reboot_notify[selinux]') }
+            it { is_expected.to_not create_kernel_parameter('enforcing') }
           end
-          let(:params) {{ ensure: 'disabled' }}
-          it { is_expected.not_to create_reboot_notify('selinux') }
-        end
-        context 'disabled -> enabled' do
-          let(:facts) do
-            os_facts = os_facts.dup
-            os_facts[:selinux] = false
-            os_facts
+          context 'ensure -> permissive' do
+            let(:facts) do
+              os_facts
+            end
+            let(:params) {{
+              ensure: 'permissive',
+              kernel_enforce: true
+            }}
+            it { is_expected.to compile.with_all_deps }
+            it { is_expected.to create_kernel_parameter('selinux').with_value(1).that_notifies('Reboot_notify[selinux]') }
+            it { is_expected.to create_kernel_parameter('enforcing').with_value(0).that_notifies('Reboot_notify[selinux]') }
           end
-          let(:params) {{ ensure: 'enforcing' }}
-          it { is_expected.to create_reboot_notify('selinux') }
-        end
-        context 'disabled -> true' do
-          let(:facts) do
-            os_facts = os_facts.dup
-            os_facts[:selinux] = false
-            os_facts
+          context 'ensure -> disabled' do
+            let(:facts) do
+              os_facts = os_facts.dup
+              os_facts[:selinux] = false
+              os_facts
+            end
+            let(:params) {{
+              ensure: 'disabled',
+              kernel_enforce: true
+            }}
+            it { is_expected.to compile.with_all_deps }
+            it { is_expected.to create_kernel_parameter('selinux').with_value(0).that_notifies('Reboot_notify[selinux]') }
+            it { is_expected.to_not create_kernel_parameter('enforcing') }
           end
-          let(:params) {{ ensure: true }}
-          it { is_expected.to create_reboot_notify('selinux') }
-        end
-        context 'disabled -> permissive' do
-          let(:facts) do
-            os_facts = os_facts.dup
-            os_facts[:selinux] = false
-            os_facts
+          context 'ensure -> enforcing' do
+            let(:facts) do
+              os_facts = os_facts.dup
+              os_facts[:selinux] = false
+              os_facts
+            end
```

Comment:
Why the duplicate tests without the os_facts (earlier in the file)?

Reply:
@sharkbruhaha They're applied (otherwise this wouldn't work). The ones down below are manipulating facts.